search-client/src/App.js

```diff
@@ -163,6 +165,7 @@ function App() {
                     icon: 'search', onClick: (event, data) => {
                       setLoading(true);

```

Comment:
I would wrap all these lines into a separate function since it's duplicated below

Reply:
I could do a lot of refactoring, will leave it to @haw230 to address

Reply:
LGTM